clients/cli/src/utils/updater.rs

```diff
@@ -0,0 +1,328 @@
+//! Core auto-updater functionality and version management
+//!
+//! This module provides the underlying implementation for:
+//! - Version tracking and persistence
+//! - Git-based version detection
+//! - Update application logic
+//! - Process management for CLI restarts
+//!
+//! The code here is used by the auto-updater thread (./updater.rs) to handle the mechanics of
+//! checking versions and applying updates in both test and production environments.
+
+use parking_lot::RwLock;
+use semver::Version;
+use std::os::unix::process::CommandExt;
+use std::sync::Arc;
+use std::{fs, process::Command};
+
+// Constants
+
+// ANSI escape codes for colors for pretty printing
+pub const BLUE: &str = "\x1b[34m"; // Normal blue
+pub const RESET: &str = "\x1b[0m"; // Reset color
+
+// The file to store the current version in
+pub const VERSION_FILE: &str = ".current_version";
+pub const REMOTE_REPO: &str = "https://github.com/nexus-xyz/network-api";
+pub const FALLBACK_VERSION: Version = Version::new(0, 3, 5); // 0.3.5
+
+#[derive(Debug, Clone, Copy, PartialEq, clap::ValueEnum)]
+pub enum AutoUpdaterMode {
+    Production,
+    Test,
+}
+
+/// Struct to manage the updater configuration
+#[derive(Clone)]
+pub struct UpdaterConfig {
+    pub mode: AutoUpdaterMode,
+    pub update_interval: u64,
+    pub repo_path: String,
+    pub remote_repo: String,
+    pub hostname: String,
+}
+
+impl UpdaterConfig {
+    pub fn new(mode: AutoUpdaterMode, hostname: String) -> Self {
+        match mode {
+            AutoUpdaterMode::Production => Self {
+                mode,
+                repo_path: format!(
+                    "{}/.nexus/network-api",
+                    std::env::var("HOME").unwrap_or_default()
+                ),
+                remote_repo: String::from("https://github.com/nexus-labs/nexus-prover.git"),
+                update_interval: 300, // check for updates every 5 minutes (300 seconds)
+                hostname,
+            },
+            AutoUpdaterMode::Test => Self {
```

Comment:
there were other modes before but got whittled down. I agree with your assessment. I will make that a different PR for that though since i may need to change a fair share.